### DISCUSSION TITLE
extra_build_args = '-parallelizeTargets ARCHS=x86_64' causes device debugging to fail on iOS 17.x

### DISCUSSION THREAD

**Discussion by nlash-livly** [wojciech-kulik/xcodebuild.nvim#164] [Q&A] [answered]
Our team's app relies on third-party libraries that cannot run on non-Rosetta simulators. When I include extra_build_args = '-parallelizeTargets ARCHS=x86_64' in the xcodebulid.nvim configuration, I am unable to debug using physical devices. The app installs, but cannot launch due to failure to verify integrity. Removing the arguments and relaunching nvim, I am able to debug using physical devices. 

Is there a way to apply/remove those extra build flags while nvim is running instead of having to update the xcodebuild.nvim config? Better yet, is there a way to exclude the flags if a physical device is selected?

**Comment by wojciech-kulik** [comment, marked as answer]
Try changing arguments before running commands using:

`require("xcodebuild.core.config").options.commands.extra_build_args = 'smth'`

You can also check `vim.g.xcodebuild_platform` to determine if it's simulator (`iOS Simulator`) or device (`iOS`)
